src/scripting/lua_gui2.cpp

```diff
@@ -280,20 +285,88 @@ int intf_add_widget_definition(lua_State* L)
 	return 0;
 }
 
+int intf_show_recruit_dialog(lua_State* L)
+{
+	const size_t len = lua_rawlen(L, 1);
+	if (!lua_istable(L, 1)) {
+		return luaL_error(L, "List of unit types not specified!");
+	}
+
+	std::vector<const unit_type*> types;
+	types.reserve(len);
+	for (size_t i = 1; i <= len; i++) {
+		lua_rawgeti(L, 1, i);
+		const unit_type* ut = luaW_tounittype(L, -1);
+		if (ut) {
+			types.push_back(ut);
+		}
+		lua_pop(L, 1);
+	}
+
+	const display* disp = display::get_singleton();
+	if (!types.empty() && disp != nullptr) {
+		gui2::dialogs::units_dialog dlg;
+		dlg.build_recruit_dialog(types, disp->playing_team());
+		if(dlg.show() && dlg.is_selected()) {
+			luaW_pushunittype(L, *types[dlg.get_selected_index()]);
+			return 1;
+		}
+	} else {
+		ERR_LUA << "Unable to show recruit dialog";
+	}
+
+	return 0;
+}
+
+int intf_show_recall_dialog(lua_State* L)
+{
+	const size_t len = lua_rawlen(L, 1);
+	if (!lua_istable(L, 1)) {
+		return luaL_error(L, "List of units not specified!");
+	}
+
+	std::vector<unit_const_ptr> units;
+	units.reserve(len);
+	for (size_t i = 1; i <= len; i++) {
+		lua_rawgeti(L, 1, i);
+		unit_const_ptr u(luaW_tounit_ptr(L, -1));
+		if (u) {
+			units.push_back(u);
+		}
+		lua_pop(L, 1);
+	}
+
+	const display* disp = display::get_singleton();
+	if (!units.empty() && disp != nullptr) {
+		gui2::dialogs::units_dialog dlg;
+		dlg.build_recall_dialog(units, disp->playing_team());
+		if(dlg.show() && dlg.is_selected()) {
+			luaW_pushunit(L, units[dlg.get_selected_index()]->underlying_id());
+			return 1;
+		}
+	} else {
+		ERR_LUA << "Unable to show recall dialog";
+	}
+
+	return 0;
+}
+
 int luaW_open(lua_State* L)
 {
 	auto& lk = lua_kernel_base::get_lua_kernel<lua_kernel_base>(L);
 	lk.add_log("Adding gui module...\n");
 	static luaL_Reg const gui_callbacks[] = {
-		{ "show_menu",          &show_menu },
-		{ "show_narration",     &show_message_dialog },
-		{ "show_popup",         &show_popup_dialog },
-		{ "show_story",         &show_story },
-		{ "show_prompt",        &show_message_box },
-		{ "show_help",          &show_help   },
-		{ "switch_theme",             &switch_theme   },
-		{ "add_widget_definition",    &intf_add_widget_definition },
-		{ "show_dialog",              &intf_show_dialog   },
+		{ "show_menu",              &show_menu },
+		{ "show_narration",         &show_message_dialog },
+		{ "show_popup",             &show_popup_dialog },
+		{ "show_story",             &show_story },
+		{ "show_prompt",            &show_message_box },
+		{ "show_help",              &show_help   },
+		{ "switch_theme",           &switch_theme },
+		{ "add_widget_definition",  &intf_add_widget_definition },
+		{ "show_dialog",            &intf_show_dialog },
+		{ "show_recruit_dialog",    &intf_show_recruit_dialog },
+		{ "show_recall_dialog",     &intf_show_recall_dialog },
```

Comment:
If these functions are only be meant to be used ingame it would be better if they'd be defined in game_lua_kernel.cpp, like gui.show_inspector, see https://github.com/wesnoth/wesnoth/blob/ebf9a2949a49d1cfaf5715bc1ec674faab6ccb4e/src/scripting/game_lua_kernel.cpp#L5257

Reply:
sure, will look into it.

Reply:
Moved to game_lua_kernel in https://github.com/wesnoth/wesnoth/pull/9499/commits/548e73b7471a3ae70d7e061f02c028fb715dc410.